### PR TITLE
Swagger API Documentation

### PR DESCRIPTION
**Pull request by evantahler**
For years, Actionhero has shipped with a `showDocumentation` action that provided a way to describe the actions running on your server. The action was used to self-document the abilities of your server. However, the format of that action is arbitrary... and not helpful in a larger ecosystem.

Switching the format of this action to swagger/OpenAPI will make the Action much more useful.

* Switch the `showDocumentation` action with `swagger` action. Remove all old references to `showDocumentation`
* Include a new `swagger.html` and `swagger` action in generated projects which can consume and demo the API
* Remove `initializer/documentation`, as we don't need this documentation internally without the old showDocumentation action. 

Screen Shot 2020-05-23 at 5 15 28 PM

The breaking change is only around the `showDocumentation` and `documentation` initializer, which appears to not be used very much.

Closes https://github.com/actionhero/actionhero/issues/1470

### CODE REVIEW

File: src/actions/showDocumentation.ts [hunk 1]
```diff
@@ -1,87 +0,0 @@
-import { api, Action } from "./../index";
```
Comment (gcoonrod) [inline src/actions/showDocumentation.ts:1]: Any reason to leave this action around for backwards compatibility?

File: src/actions/swagger.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+          return;
+        }
+
+        swaggerPaths[formattedPath] = swaggerPaths[formattedPath] || {};
+        swaggerPaths[formattedPath][method] = {
+          tags: [tag],
+          summary: action.description,
+          // description: action.description, // this is redundant
+          consumes: ["application/json"],
+          produces: ["application/json"],
```
Comment (gcoonrod) [inline src/actions/swagger.ts:78]: Do our actions _always_ consume and produce `application/json`?

File: src/actions/swagger.ts [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+                type: "string", // not really true, but helps the swagger validator
+                required:
+                  action.inputs[inputName].required ||
+                  route.path.includes(`:${inputName}`)
+                    ? true
+                    : false,
+                default:
+                  action.inputs[inputName].default !== null &&
+                  action.inputs[inputName].default !== undefined
+                    ? typeof action.inputs[inputName].default === "object"
```
Comment (gcoonrod) [inline src/actions/swagger.ts:94]: Typescript doesn't give us something better than `typeof of === "object"`?

**Comment by evantahler** [comment]
Great questions @gcoonrod, and thanks for the review!  

I think that this action is a lot like `status` - you are probably going to want to customize it for your application.  Does you API only speak `application/json`?   Do you want to add tags?  Do you want to hide duplicate routes?  No idea!  I think I was trying to start with a 'common enough' boilerplate that would get folks started.  That's why I like having this logic contained entirely in an action (userland) vs in an initializer deep within Actionhero that would be harder to customize. 

As far as backwards-compatibility goes, I got the sense that no one was really using the `documentation` actions/initializers... but I'll ask again in Slack!

File: src/actions/swagger.ts (hunk 3 above)
Comment (evantahler) [reply src/actions/swagger.ts:94]: not that I've found yet :(

File: src/actions/swagger.ts [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+      //   return { name: tag, description: `topic: ${tag}` };
+      // }),
+      schemes: config.servers.web.allowedRequestHosts[0]
+        ? ["https", "http"]
+        : ["http"],
+      paths: swaggerPaths,
+
+      securityDefinitions: {
+        // TODO (custom)?
+      },
```
Comment (evantahler) [inline src/actions/swagger.ts:140]: Here's an example https://github.com/grouparoo/grouparoo/blob/master/core/api/src/actions/swagger.ts#L134-L145

**Pull request merged**